### PR TITLE
[WIP] tests refactor: separate connector-specific tests

### PR DESCRIPTION
**Pull request by findepi**
This implements a partial cleanup of connector-specific and connector-independent tests.
It's targeted at `starburstdata/presto`, since there is no intention to merge this, just to gather feedback.

What this does:
1. moves selected connector-specific tests to `AbstractTestIntegrationSmokeTest` (this seems to be _the class_ dedicated to connector-specific tests, despite the name)

What this does not:
1. This does not introduce any mechanism for bulk disabling of tests. Still, if a specific class is not expected to run a test (e.g. `testCreateTable` because e.g. table creation is not supported), the test method needs to be overridden.

What is still to be done:
1. Move other connector-specific tests from `AbstractTestDistributedQueries` in the same manner (rename table; add column; drop column; insert; view-related; testWrittenStats?)
2. remove connector-specific subclasses of `AbstractTestDistributedQueries`

### CODE REVIEW

**Comment by findepi** [comment]
cc @kokosing @sopel39 @losipiuk

File: presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestIntegrationSmokeTest.java [hunk 1]
```diff
@@ -224,6 +224,7 @@ private MaterializedResult getExpectedOrdersTableDescription(boolean dateSupport
         }
     }
 
+    // This also tests DROP TABLE
     @Test
     public void testCreateTable()
```
Comment (kokosing) [inline presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestIntegrationSmokeTest.java]: maybe rename the test method

**Review by kokosing — COMMENTED** [review]
so far so good.

File: presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestIntegrationSmokeTest.java (hunk 1 above)
Comment (findepi) [reply presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestIntegrationSmokeTest.java]: yea... but `testCreateAndDrop` sounds so strange...

**Comment by findepi** [comment]
Some commits extracted to https://github.com/prestodb/presto/pull/11045

**Pull request closed without merging**